src/main/java/harmonised/pmmo/client/gui/XPOverlayGUI.java

```diff
@@ -64,68 +64,83 @@ public void render(ForgeIngameGui gui, PoseStack stack, float partialTick, int w
 	private LinkedHashMap<String, SkillLine> lineRenderers = new LinkedHashMap<>();
 
-	private void renderSkillList(PoseStack stack, int skillListX, int skillListY)
+	private void renderSkillList(com.mojang.blaze3d.platform.Window window, PoseStack stack, float offsetX, float offsetY)
 	{
+		int posX = (int) Math.round(offsetX * window.getGuiScaledWidth());
+		int posY = (int) Math.round(offsetY * window.getGuiScaledHeight());
+		int lineHeight = fontRenderer.lineHeight;
 		if (ClientTickHandler.isRefreshTick()) {
 			modifiers = core.getConsolidatedModifierMap(mc.player);	
 			skillsKeys = core.getData().getXpMap(null).keySet().stream()
 					.sorted(Comparator.<String>comparingLong(a -> core.getData().getXpRaw(null, a)).reversed())
 					.toList();
 			var holderMap = lineRenderers;
 			lineRenderers.clear();
-			AtomicInteger yOffset = new AtomicInteger(0);
+			AtomicInteger line = new AtomicInteger(0);
 			skillGap = skillsKeys.stream()
 					.map(skill -> fontRenderer.width(new TranslatableComponent("pmmo."+skill).getString()))
 					.max(Comparator.comparingInt(t -> t)).orElse(0);
 			skillsKeys.forEach((skillKey)-> {
 				var xpRaw = core.getData().getXpRaw(null, skillKey);
 				lineRenderers.put(skillKey, 
 					xpRaw != holderMap.getOrDefault(skillKey, SkillLine.DEFAULT).xpValue()
-					? new SkillLine(skillKey, modifiers.getOrDefault(skillKey, 1.0), xpRaw, yOffset.get(), skillGap)
-					: new SkillLine(holderMap.get(skillKey), yOffset.get()));
-				yOffset.getAndIncrement();
+					? new SkillLine(skillKey, modifiers.getOrDefault(skillKey, 1.0), xpRaw, line.get(), lineHeight, skillGap)
+					: new SkillLine(holderMap.get(skillKey), line.get(), lineHeight));
+				line.getAndIncrement();
 			});
 		}
 		
 		lineRenderers.forEach((skill, line) -> {
-			line.render(stack, skillListX, skillListY, fontRenderer);
+			line.render(stack, posX, posY, fontRenderer);
 		});
 	}
 	
 	private int maxCharge = 0;
 	private int currentCharge = 0;
 	
-	private void renderVeinGauge(PoseStack stack, int gaugeX, int gaugeY) {
+	private void renderVeinGauge(com.mojang.blaze3d.platform.Window window, PoseStack stack, float offsetX, float offsetY) {
+		int posX = (int) Math.round(offsetX * window.getGuiScaledWidth());
+		int posY = (int) Math.round((1.0 - offsetY) * window.getGuiScaledHeight());
+		int lineHeight = (int) Math.round(fontRenderer.lineHeight * 1.2);
 		if (ClientTickHandler.isRefreshTick()) {
 			maxCharge = VeinMiningLogic.getMaxChargeFromAllItems(mc.player);
 			if (maxCharge > 0)
 				currentCharge = VeinTracker.getCurrentCharge();
 		}
-		if (currentCharge > 0) {
-			GuiComponent.drawString(stack, fontRenderer, LangProvider.VEIN_LIMIT.asComponent(Config.VEIN_LIMIT.get()), gaugeX, gaugeY-11, 0xFFFFFF);
-			GuiComponent.drawString(stack, fontRenderer, LangProvider.VEIN_CHARGE.asComponent(currentCharge, maxCharge), gaugeX, gaugeY, 0xFFFFFF);
+		int percentCharged = (int) Math.round(((float) currentCharge / (float) maxCharge) * 100);
```

Comment:
This is counterintuitive to the design of the vein miner.  The value on the left is how much charge is accrued.  if you were to unequip all of your items, you would still have this charge accrued (we call it overcharging).  if this were a percent, it would not only divide by zero but not show the user a number that means anything to them.